src/main/kotlin/com/terraformation/backend/search/SearchService.kt

```diff
@@ -181,4 +182,35 @@ class SearchService(private val dslContext: DSLContext) {
     // way of filtering out.
     return searchResults.map { it?.get(fieldPathName)?.toString() }.distinct()
   }
+
+  fun searchCount(
+      rootPrefix: SearchFieldPrefix,
+      fields: Collection<SearchFieldPath>,
+      criteria: Map<SearchFieldPrefix, SearchNode>,
+  ): Long {
+    val exactCriteria = criteria.mapValues { it.value.toExactSearch() }
+    if (exactCriteria != criteria) {
+      val exactResults = searchCount(rootPrefix, fields, exactCriteria)
+      if (exactResults > 0) {
+        return exactResults
+      }
+    }
+
+    return runQueryCount(rootPrefix, fields, criteria)
+  }
+
+  private fun runQueryCount(
+      rootPrefix: SearchFieldPrefix,
+      fields: Collection<SearchFieldPath>,
```

Comment:
Do we even need a field list for this?

Reply:
Agreed, though I wouldn't treat it as super high priority because I believe in practice we don't use flattened sublists in a way that would cause any behavior differences here. But it'd be better to either make it work properly or at least detect a potentially problematic query and throw an error.